src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModification.java

```diff
@@ -82,38 +70,63 @@ public void apply(Network network, Reporter subReporter) {
                 // we retrieve the absolute value of susceptance per section, then we determine the sign using the type
                 double bPerSectionAbsoluteValue = Math.abs(model.getBPerSection());
                 double newBPerSection = shuntCompensatorType == ShuntCompensatorType.CAPACITOR ? bPerSectionAbsoluteValue : -bPerSectionAbsoluteValue;
                 model.setBPerSection(newBPerSection);
             }
         }
 
-        if (modificationInfos.getQAtNominalV() != null) {
+        if (modificationInfos.getMaxQAtNominalV() != null) {
             double olQAtNominalV = Math.abs(Math.pow(voltageLevel.getNominalV(), 2) * model.getBPerSection());
-            double susceptancePerSection = modificationInfos.getQAtNominalV().getValue() / Math.pow(voltageLevel.getNominalV(), 2);
-
+            double newQatNominalV = modificationInfos.getMaxQAtNominalV().getValue() / maximumSectionCount;
+            double susceptancePerSection = newQatNominalV / Math.pow(voltageLevel.getNominalV(), 2);
+            reports.add(ModificationUtils.getInstance().buildModificationReport(olQAtNominalV, newQatNominalV, "Q at nominal voltage"));
             model.setBPerSection(shuntCompensatorType == ShuntCompensatorType.CAPACITOR ? susceptancePerSection : -susceptancePerSection);
-            reports.add(ModificationUtils.getInstance().buildModificationReport(olQAtNominalV, modificationInfos.getQAtNominalV().getValue(), "Q at nominal voltage"));
-
-            AttributeModification<Integer> sectionCountModification = AttributeModification.toAttributeModification(modificationInfos.getQAtNominalV().getValue() == 0. ? 0 : 1, OperationType.SET);
-            reports.add(ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(shuntCompensator::setSectionCount, shuntCompensator::getSectionCount, sectionCountModification, "Section count"));
         }
 
-        if (modificationInfos.getSusceptancePerSection() != null) {
-            reports.add(ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(model::setBPerSection, model::getBPerSection, modificationInfos.getSusceptancePerSection(), "Susceptance per section"));
-
-            AttributeModification<Integer> sectionCountModification = AttributeModification.toAttributeModification(modificationInfos.getSusceptancePerSection().getValue() == 0. ? 0 : 1, OperationType.SET);
-            reports.add(ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(shuntCompensator::setSectionCount, shuntCompensator::getSectionCount, sectionCountModification, "Section count"));
+        if (modificationInfos.getMaxSusceptance() != null) {
+            double susceptancePerSection = modificationInfos.getMaxSusceptance().getValue() / maximumSectionCount;
+            reports.add(ModificationUtils.getInstance().buildModificationReport(model.getBPerSection(), susceptancePerSection, "Susceptance per section"));
```

Comment:
Shouldn't the logs be now about qmax at nominal voltage and max susceptance ?

Reply:
I agree, Done